time_unit/__init__.py

```
@@ -249,7 +256,13 @@ def _get_range(cls, item):
             return item, item
         elif isinstance(item, Timeunit):
             return item.date_range
-        raise TypeError('Item {item!r} has no date range.')
+        # try to make a range
+        try:
+            dt0, dt1 = item
+            if isinstance(dt0, date) and isinstance(dt1, date):
+                return item
+        except TypeError:
+            raise TypeError('Item {item!r} has no date range.')
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)